Mesh_3/include/CGAL/Mesh_3/Construct_initial_points_labeled_image.h

```diff
@@ -0,0 +1,333 @@
+          const Index& intersect_index = std::get<1>(intersect);
+          // [get construct intersection]
+          Weighted_point pi(intersect_point);
+
+          // This would cause trouble to optimizers
+          // check pi will not be hidden
+          typename Tr::Locate_type lt;
+          int li, lj;
+          Cell_handle pi_cell = tr.locate(pi, lt, li, lj);
+          if(lt != Tr::OUTSIDE_AFFINE_HULL) {
+            switch (tr.dimension())
+            { //skip dimension 0
+            case 1:
+              if (tr.side_of_power_segment(pi_cell, pi, true) != CGAL::ON_BOUNDED_SIDE)
+                continue;
+              break;
+            case 2:
+              if (tr.side_of_power_circle(pi_cell, 3, pi, true) != CGAL::ON_BOUNDED_SIDE)
+                continue;
+              break;
+            case 3:
+              if (tr.side_of_power_sphere(pi_cell, pi, true) != CGAL::ON_BOUNDED_SIDE)
+                continue;
+            }
+          }
+
+          //check pi is not inside a protecting ball
+          std::vector<Vertex_handle> conflict_vertices;
+          if (tr.dimension() == 3)
+          {
+            tr.vertices_on_conflict_zone_boundary(pi, pi_cell
+              , std::back_inserter(conflict_vertices));
+          }
+          else
+          {
+            for (typename Tr::Finite_vertices_iterator vit = tr.finite_vertices_begin();
+                 vit != tr.finite_vertices_end(); ++vit)
+            {
+              const Weighted_point& wp = tr.point(vit);
+              if (cwsr(wp, FT(0)) == CGAL::SMALLER) // 0 < wp's weight
+                conflict_vertices.push_back(vit);
+            }
+          }
+
+          bool pi_inside_protecting_sphere = false;
+          for(Vertex_handle cv : conflict_vertices)
+          {
+            if(tr.is_infinite(cv))
+              continue;
+
+            const Weighted_point& cv_wp = tr.point(cv);
+            if (cwsr(cv_wp, FT(0)) == CGAL::EQUAL) // 0 == wp's weight
+              continue;
+
+            // if the (squared) distance between intersect_point and cv is smaller or equal than cv's weight
+            if (cwsr(cv_wp, - tr.min_squared_distance(intersect_point, cp(cv_wp))) != CGAL::LARGER)
+            {
+              pi_inside_protecting_sphere = true;
+              break;
+            }
+          }
+          if (pi_inside_protecting_sphere)
+            continue;
+
+          // insert point in temporary triangulation
+
+          /// The following lines show how to insert initial points in the
+          /// `c3t3` object. [insert initial points]
+          Vertex_handle v = tr.insert(pi);
+
+          // `v` could be null if `pi` is hidden by other vertices of `tr`.
+          CGAL_assertion(v != Vertex_handle());
+
+          c3t3.set_dimension(v, 2); // by construction, points are on surface
+          c3t3.set_index(v, intersect_index);
+          /// [insert initial points]
+
```

Comment:
Probably line 311 has to be split and maybe even a "trick" with `\noop` should be used so instead of 
```
/// [insert initial points]
```
the line
```
/// \noop [insert initial points]
```
should be used in both cases.